### PR TITLE
[#914] Create span for CredentialsService.get

### PR DESCRIPTION
**Pull request by calohmn**
This is for #914.

### CODE REVIEW

**Comment by calohmn** [comment]
I've rebased and adapted the commit to the recent changes for the vertx update.
(Failing CI build doesn't seem caused by changes here.)

I'm about to create a 2nd PR that will add the necessary changes to the `CredentialsClient`.

**Comment by calohmn** [comment]
I've applied the signature change in `newChildSpan` and have refactored the `BaseCredentialsService.processGetRequest()` method.
Concerning the query parameters: I prefer having them as Tags, also to enable searching for them.

File: service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java [hunk 1]
```diff
@@ -80,47 +93,110 @@ protected String getEventBusAddress() {
...
+            TracingHelper.logError(span, "missing tenant and/or payload");
+            resultFuture = Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
         } else {
             final String type = removeTypesafeValueForField(String.class, payload, CredentialsConstants.FIELD_TYPE);
             final String authId = removeTypesafeValueForField(String.class, payload,
                     CredentialsConstants.FIELD_AUTH_ID);
             final String deviceId = removeTypesafeValueForField(String.class, payload,
                     CredentialsConstants.FIELD_PAYLOAD_DEVICE_ID);
 
+            span = newChildSpan(SPAN_NAME_GET_CREDENTIALS, request.getSpanContext(), tenantId);
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java]: there doesn't seem to be a difference between the *if* block and the *else* block regarding the creation of the span, is there?

File: service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java [hunk 2]
```diff
@@ -80,47 +93,110 @@ protected String getEventBusAddress() {
...
+                TracingHelper.logError(span, String.format(
+                        "invalid search criteria [type: %s, device-id: %s, auth-id: %s]", type, deviceId, authId));
                 log.debug("get credentials request contains invalid search criteria [type: {}, device-id: {}, auth-id: {}]",
                         type, deviceId, authId);
-                return Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
+                resultFuture = Future.failedFuture(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST));
+            }
+        }
+        return resultFuture.compose(eventBusMessage -> {
+            Tags.HTTP_STATUS.set(span, eventBusMessage.getStatus());
```
Comment (sophokles73) [inline service-base/src/main/java/org/eclipse/hono/service/credentials/BaseCredentialsService.java]: I think we'll want to set this tag in any case, right?

**Pull request merged**